ras/ras_lsvpd.py

```diff
@@ -60,6 +62,50 @@ def run_cmd_out(cmd):
                                      ignore_status=True,
                                      sudo=True).decode("utf-8").strip()
 
+    def test_build_upstream(self):
+        """
+        For upstream target download and compile source code
+        Caution : This function will overwrite system installed
+        lsvpd Tool binaries with upstream code.
+        """
+        if self.run_type == 'upstream':
+            self.detected_distro = distro.detect()
+            deps = ['gcc', 'make', 'automake', 'autoconf', 'bison', 'flex',
+                    'libtool', 'zlib-devel', 'ncurses-devel', 'librtas-devel']
+            if 'SuSE' in self.detected_distro.name:
+                deps.extend(['libsgutils-devel', 'sqlite3-devel',
+                             'libvpd2-devel'])
+            elif self.detected_distro.name in ['centos', 'fedora', 'rhel']:
+                deps.extend(['sqlite-devel', 'libvpd-devel'])
+            else:
+                self.cancel("Unsupported Linux distribution")
+            for package in deps:
+                if not self.sm.check_installed(package) and not \
+                        self.sm.install(package):
+                    self.cancel("Fail to install %s required for this test." %
+                                package)
+            url = self.params.get(
+                'lsvpd_url', default='https://github.com/power-ras/'
+                'lsvpd/archive/refs/heads/master.zip')
+            tarball = self.fetch_asset('lsvpd.zip', locations=[url],
+                                       expire='7d')
+            archive.extract(tarball, self.workdir)
+            self.sourcedir = os.path.join(self.workdir, 'lsvpd-master')
+            os.chdir(self.sourcedir)
+            self.run_cmd('./bootstrap.sh')
+            # TODO : For now only this test is marked as failed.
+            # Additional logic should be added to skip all the remaining
+            # test_() functions for upstream target if source code
+            # compilation fails. This will require a way to share
+            # variable/data across test_() functions.
+            self.run_cmd('./configure --prefix=/usr')
+            if self.is_fail >= 1:
+                self.fail("Source code compilation error")
+            build.make(self.sourcedir)
+            build.make(self.sourcedir, extra_args='install')
+        else:
+            self.cancel("This test is supported with upstream as a target")
```

Comment:
is this `not supported with` ?

Reply:
the message is correct. Test is supported with upstream as a target. for bistro as a target it will be cancelled.

Reply:
ok